src/Compilers/Core/Portable/Emit/EditAndContinue/SynthesizedTypeMaps.cs

```diff
@@ -35,8 +34,8 @@ public bool IsEmpty
     /// A map of the assembly identities of the baseline compilation to the identities of the original metadata AssemblyRefs.
     /// Only includes identities that differ between these two.
     /// </summary>
-    public ImmutableSegmentedDictionary<string, AnonymousTypeValue> AnonymousDelegatesWithIndexedNames { get; }
-        = anonymousDelegatesWithIndexedNames ?? ImmutableSegmentedDictionary<string, AnonymousTypeValue>.Empty;
+    public ImmutableSegmentedDictionary<AnonymousDelegateWithIndexedNamePartialKey, ImmutableArray<AnonymousTypeValue>> AnonymousDelegatesWithIndexedNames { get; }
```

Comment:
>AnonymousDelegateWithIndexedNamePartialKey [](http://example.com/codeflow?start=40&length=42)

The dictionary key was the unique synthesized delegate name, and is now just the type parameter count and parameter count. Are we no longer able to use the unique name?

Reply:
We would only be able to use the unique name if it was entirely deterministically generated based on the signature (e.g. some sort of hash of the signature types).

Reply:
The same actually applies to anonymous types - if we hashed the names of their fields instead of using an ordinal in the template name.